File: ecommerce-frontend/src/api/review.ts
import apiClient from './client';

export interface ReviewUser {
  id: number;
  username: string;
  full_name: string | null;
}

export interface ReviewItem {
  id: number;
  rating: number;
  comment: string | null;
  created_at: string;
  user: ReviewUser | null;
}

export interface ReviewStats {
  average: number;
  count: number;
  distribution: Record<string, number>;
}

export const reviewApi = {
  list: async (productId: number): Promise<ReviewItem[]> => {
    const res = await apiClient.get(`/reviews/product/${productId}`);
    return res.data;
  },

  stats: async (productId: number): Promise<ReviewStats> => {
    const res = await apiClient.get(`/reviews/product/${productId}/stats`);
    return res.data;
  },

  create: async (productId: number, data: { rating: number; comment?: string }): Promise<ReviewItem> => {
    const res = await apiClient.post(`/reviews/product/${productId}`, data);
    return res.data;
  },
};

File: ecommerce-backend/app/schemas/review.py
# app/schemas/review.py
from pydantic import BaseModel, Field
from datetime import datetime


class ReviewCreate(BaseModel):
    rating: int = Field(..., ge=1, le=5)
    comment: str | None = Field(None, max_length=500)


class ReviewUserOut(BaseModel):
    id: int
    username: str
    full_name: str | None = None

    class Config:
        from_attributes = True


class ReviewOut(BaseModel):
    id: int
    rating: int
    comment: str | None = None
    created_at: datetime
    user: ReviewUserOut | None = None

    class Config:
        from_attributes = True


class ReviewStats(BaseModel):
    average: float = 0
    count: int = 0
    distribution: dict[str, int] = {}  # {"5": 3, "4": 1, ...}
